bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v3/ImageImpl.java

```diff
@@ -161,14 +168,65 @@ public String getSrcset() {
 
     @Nullable
     @Override
-    public String getWidth() {
-        return width;
+    @JsonIgnore
+    public String[] getBaseImageResolution() {
+        String baseImageAbsoluteUrl = externalizer.publishLink(resourceResolver, super.getSrc());
+        URL url;
+        InputStream in;
+        BufferedImage image = null;
+        try {
+            url = new URL(baseImageAbsoluteUrl);
```

Comment:
We can load the file locally and compute the original width and height, no need to do a HTTP request. For DAM assets we can even leverage `new EnhancedRendition(asset.getOriginal())`

Reply:
The image dimensions should be saved in the repository to avoid having to compute it for each request.

Reply:
Implemented `getOriginalDimension` for now to get the image dimensions only if the image is a DAM asset.